tests/app/test_jinja_filters.py

```diff
@@ -1008,3 +1005,52 @@ def test_format_address_list_concatenated_list_no_values(self):
             format_address_list(answer_address, metadata_address)
 
         self.assertEqual('No valid address passed to format_address_list filter', error.exception.args[0])
+
+    def test_first_non_empty_item_filter_returns_first_non_empty_item(self):
+        # Given
+        expected_actual_scenarios = [
+            ['only value', (Markup('only value'),)],
```

Comment:
You could use [pytest parameterize](https://docs.pytest.org/en/latest/parametrize.html) here. Not worth changing, but worth bearing in mind in future.

Reply:
Ah this is exactly what I needed.

Reply:
Not sure if it will actually work until we remove the unittest.TestCase inheritance.